src/core/Akka/Actor/CoordinatedShutdown.cs

```diff
@@ -653,13 +653,15 @@ internal static void InitPhaseActorSystemTerminate(ActorSystem system, Config co
 
                     if (terminateActorSystem)
                     {
-                        system.FinalTerminate();
-                        return system.Terminate().ContinueWith(tr =>
```

Comment:
This potentially an infinite loop call, if the call came from `ActorSystem.Terminate()` in the first place.